v3/cmd/zlint/main.go

```diff
@@ -195,18 +191,14 @@ func doLint(inputFile *os.File, inform string, registry lint.Registry) {
 		log.Fatalf("unknown input format %s", format)
 	}
 	var zlintResult *zlint.ResultSet
-	if isCRL {
-		crl, err := x509.ParseRevocationList(asn1Data)
-		if err != nil {
-			log.Fatalf("unable to parse certificate revocation list: %s", err)
-		}
+	if cert, err := x509.ParseCertificate(asn1Data); err == nil {
+		zlintResult = zlint.LintCertificateEx(cert, registry)
+	} else if crl, err := x509.ParseRevocationList(asn1Data); err == nil {
 		zlintResult = zlint.LintRevocationListEx(crl, registry)
+	} else if resp, err := ocsp.ParseResponse(asn1Data, nil); err == nil {
+		zlintResult = zlint.LintOcspResponseEx(resp, registry)
 	} else {
-		c, err := x509.ParseCertificate(asn1Data)
-		if err != nil {
-			log.Fatalf("unable to parse certificate: %s", err)
-		}
-		zlintResult = zlint.LintCertificateEx(c, registry)
+		log.Fatalf("unable to parse input as a certificate, CRL, or OCSP response")
```

Comment:
Would mind reporting the error as well in order to reduce debugging cycles for consumers?

Reply:
I have updated to report error in case we fail to parse